onesixtyone.c

```diff
@@ -142,7 +142,11 @@ void read_communities(char* filename)
     printf("MAX_COMMUNITIES (%d) reached. Remaining communities will be skipped \n", i);
   }
 
-  community_count = i;
+  if (c == 0 && ch == -1)
+	community_count = i - 1;
```

Comment:
Does this do the right thing when the community file is completely empty? If I'm reading the code correctly, `i` will when be `0`, meaning that `community_count == -1` and we'll likely end up crashing elsewhere when we index with it.

Reply:
I'll test, but it might be a problem I didn't think about.

If the file has no characters, just a `'\n'`, we should be okay.

Reply:
Yep -- it's more the case where the file is totally empty that I'm worried about. With just a newline I expect it'll work correctly.

Reply:
good news, it doesn't have a problem with empty files.

It enters the first for-loop (var `i`), reads a char, doesn't enter the second for-loop (var `c`), increments `i` and exits the first foor-loop. So `community_count` receives `i - 1` which is `0`.

Reply:
Yep, looks like you're right. Thanks for checking!